### DISCUSSION TITLE
ZIP-PAB01 energy meter module does not update sensor values.

### DISCUSSION THREAD

**Discussion by clase30** [zwave-js/zwave-js#6600] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
ZIP-PAB01 energy meter module does not update sensor values. The module sends the value (reflected in the log) every time it detects a 5% variation in consumption, but Home Assistant does not change the value of the sensors, they remain frozen

What did you expect to happen instead?
Hopefully the values ​​will reflect the changes, for example in the settings -> devices window

  
Anything else we should know?
It is a bug detected and reported by several users. For example:
https://forum.hacf.fr/t/pab01-pas-de-remonter-automatique-des-valeurs/15028

### Software versions

Driver (node-zwave-js): ...  
Versión del controlador: 12.3.0

Z-Wave JS UI: ...
zwave-js-ui: 9.3.2

Home Assistant Z-Wave Integration: ...
Versión del servidor: 1.33.0

Home Assistant Z-Wave JS Addon: ...
ioBroker.zwave2 Adapter: ...
If you are using something non-standard, tell us here: ...

### Device information

Manufacturer: ...
Philio Technology Corp / Zipato Micromodule Energy Meter

Model name: ...
PAB01 Zipato Micromodule Energy Meter

Node ID: ... 
Node 034

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-01-10.log](https://github.com/zwave-js/node-zwave-js/files/13906519/zwavejs_2024-01-10.log)

**Comment by AlCalzone** [comment]
This looks like an issue in Home Assistant, or the sensor is hidden/disabled. `node-zwave-js` is receiving the changes and emitting the corresponding events, or you wouldn't have these lines in your log:
```
[Node 034] [~] [Meter] value[66049]: 410.1 => 408.7
```

Side note: I'd advise against reporting energy consumption so frequently: https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=optimizing-the-reporting-configuration

**Reply by clase30** [reply]
Values ​​not reaching device in z-wave integration
Where do you think the error could be, to report it?

-Issue tracker Home Assistant Core
-Issue tracker Home Assistant Frontend
-Issue tracker Home Assistant Supervisor
-Issue tracker Home Assistant Operating System

![HA disp](https://github.com/zwave-js/node-zwave-js/assets/155466412/8de494e7-6d7c-457b-99ff-7107d883d6c5)

**Reply by AlCalzone** [reply]
Home Assistant Core if I'm not mistaken.

**Comment by kpine** [comment]
I'm coming from https://github.com/home-assistant/core/issues/108878, this is an endpoint configuration problem in the driver, not an HA problem.

First of all, in the provided log the device only reports values for `Electric Consumption [W]`, so either it never updates the other values or the logs aren't showing it. That would be a device problem, not HA, as obviously HA cannot update entities if there are no reports. If it isn't reporting anything you may need to poll it. 

Second, the device only reports on Endpoint 0 and Endpoint 1:

```text
...
2024-01-10T21:33:58.984Z CNTRLR   [Node 031] [Meter] value[66049]: metadata updated                 [Endpoint 1]
2024-01-10T21:33:58.989Z CNTRLR   [Node 031] [~] [Meter] value[66049]: 0 => 0                       [Endpoint 1]
...
2024-01-10T21:34:55.390Z CNTRLR   [Node 034] [Meter] value[66049]: metadata updated                 [Endpoint 0]
2024-01-10T21:34:55.392Z CNTRLR   [Node 034] [~] [Meter] value[66049]: 378.7 => 327                 [Endpoint 0]
```

But Z-Wave JS has mapped the meter reports to endpoints 1-3. HA is never told about an endpoint 0 so it can't handle the updates.

Here are the entity configurations for Electric Consumption [W]:

```json
      {
        "domain": "sensor",
        "entity_id": "sensor.contador_energia_electric_consumption_w",
        "original_name": "Electric Consumption [W]",
        "original_device_class": "power",
        "disabled": false,
        "disabled_by": null,
        "hidden_by": null,
        "original_icon": null,
        "entity_category": null,
        "supported_features": 0,
        "unit_of_measurement": "W",
        "value_id": "35-50-1-value-66049",
        "primary_value": {
          "command_class": 50,
          "command_class_name": "Meter",
          "endpoint": 1,
          "property": "value",
          "property_name": "value",
          "property_key": 66049,
          "property_key_name": "Electric_W_Consumed"
        }
      },
      {
        "domain": "sensor",
        "entity_id": "sensor.contador_energia_electric_consumption_w_2",
        "original_name": "Electric Consumption [W] (2)",
        "original_device_class": "power",
        "disabled": false,
        "disabled_by": null,
        "hidden_by": null,
        "original_icon": null,
        "entity_category": null,
        "supported_features": 0,
        "unit_of_measurement": "W",
        "value_id": "35-50-2-value-66049",
        "primary_value": {
          "command_class": 50,
          "command_class_name": "Meter",
          "endpoint": 2,
          "property": "value",
          "property_name": "value",
          "property_key": 66049,
          "property_key_name": "Electric_W_Consumed"
        }
      },
      {
        "domain": "sensor",
        "entity_id": "sensor.contador_energia_electric_consumption_w_3",
        "original_name": "Electric Consumption [W] (3)",
        "original_device_class": "power",
        "disabled": false,
        "disabled_by": null,
        "hidden_by": null,
        "original_icon": null,
        "entity_category": null,
        "supported_features": 0,
        "unit_of_measurement": "W",
        "value_id": "35-50-3-value-66049",
        "primary_value": {
          "command_class": 50,
          "command_class_name": "Meter",
          "endpoint": 3,
          "property": "value",
          "property_name": "value",
          "property_key": 66049,
          "property_key_name": "Electric_W_Consumed"
        }
```

There is no endpoint 0 provided. Here are the value IDs provided to HA by the driver:

```json
        "35-50-1-value-66049": {
          "endpoint": 1,
          "commandClass": 50,
          "commandClassName": "Meter",
          "property": "value",
          "propertyKey": 66049,
          "propertyName": "value",
          "propertyKeyName": "Electric_W_Consumed",
          "ccVersion": 3,
          "metadata": {
            "type": "number",
            "readable": true,
            "writeable": false,
            "label": "Electric Consumption [W]",
            "ccSpecific": {
              "meterType": 1,
              "scale": 2,
              "rateType": 1
            },
            "unit": "W",
            "stateful": true,
            "secret": false
          },
          "value": 349.6
        },
...
        "35-50-2-value-66049": {
          "endpoint": 2,
          "commandClass": 50,
          "commandClassName": "Meter",
          "property": "value",
          "propertyKey": 66049,
          "propertyName": "value",
          "propertyKeyName": "Electric_W_Consumed",
          "ccVersion": 3,
          "metadata": {
            "type": "number",
            "readable": true,
            "writeable": false,
            "label": "Electric Consumption [W]",
            "ccSpecific": {
              "meterType": 1,
              "scale": 2,
              "rateType": 1
            },
            "unit": "W",
            "stateful": true,
            "secret": false
          },
          "value": 0
        },
...
        "35-50-3-value-66049": {
          "endpoint": 3,
          "commandClass": 50,
          "commandClassName": "Meter",
          "property": "value",
          "propertyKey": 66049,
          "propertyName": "value",
          "propertyKeyName": "Electric_W_Consumed",
          "ccVersion": 3,
          "metadata": {
            "type": "number",
            "readable": true,
            "writeable": false,
            "label": "Electric Consumption [W]",
            "ccSpecific": {
              "meterType": 1,
              "scale": 2,
              "rateType": 1
            },
            "unit": "W",
            "stateful": true,
            "secret": false
          },
          "value": 352.5
        },
...
```

There is no endpoint 0 value ID provided, so HA doesn't handle any reports for Endpoint 0. Thus the driver needs to properly provide a value ID for endpoint 0, or configure the endpoint reports in some other way such that they map to endpoints 1-3.

**Reply by kpine** [reply]
Also, make sure you have re-interviewed the device if it was included in an older version of Z-Wave JS, just to make sure you aren't being affected by any prior issues.

**Reply by AlCalzone** [reply]
Second the re-interview. Seems that was never done completely:
```
Node 31, Endpoint 1: Trying to access endpoint instance before Multi Channel interview
```

@clase30 please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug`, re-interview the device, and once that is done, attach the log here as a file (drag & drop into the text field).

The config file for this device does not define any mapping, although we might have to preserve the root endpoint.

**Comment by clase30** [comment]
[zwave-js-ui-store (2).zip](https://github.com/zwave-js/node-zwave-js/files/14129191/zwave-js-ui-store.2.zip)

Here is the requested log. 
It is NODE 035 

(I tried a few days ago excluding and then including this device, also interviewed. Still not updating values. It is a common problem for many users of the device.) Look for node 035.

**Reply by AlCalzone** [reply]
Ok, I took a look at the log and the manual (which isn't super clear about this).
The device has 3 endpoints, I assume EP 1 is the sum, EP2 the first channel and EP3 the second channel. The root endpoint (0) is likely a mirror of endpoint 1, which is why Z-Wave JS hides it by default.

The way the associations are configured right now, the device reports the sum of channels 1+2, and only from endpoint 0, so the reports are not exposed to applications because that endpoint is hidden.
We might have to play around with this a bit, since Z-Wave associations are a mess and the correct setup isn't always possible to detect automatically. 

I **think** the following is a good first attempt - You'll have to use Z-Wave JS UI to do this:
- Change the association in group 1 to use target endpoint 0 (root endpoint) instead of None
- Add the same association to groups 2 and 3, also both with target endpoint 0 (root endpoint).
- DO NOT re-interview afterwards

Now monitor the incoming value updates and ideally log it too, so I can take another look if this doesn't work.

**Comment by clase30** [comment]
Ok, wow! Now it is working fine !!  The values ​​are all updated as consumption varies.

![Captura de pantalla 2024-02-04 215551](https://github.com/zwave-js/node-zwave-js/assets/155466412/0e9c91af-406d-499c-ac2a-46abdfbc0ddf)
[zwave-js-ui-store (3).zip](https://github.com/zwave-js/node-zwave-js/files/14172332/zwave-js-ui-store.3.zip)
